app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -47,7 +47,7 @@ def custom_data
   end
 
   def prepare_for_clone_task
-    nic_id = create_nic
+    nic_id = valid_floating_ip? ? associated_nic : create_nic
```

Comment:
This could be simplified to:
```ruby
nic_id = associated_nic || create_nic

... and ....

def associated_nic
  floating_ip.try(:network_port).try(:ems_ref)
end
```